### ISSUE TITLE
FFmpegReader test fails, the output file has no audio

### ISSUE THREAD

**Issue by steils** [OpenShot/libopenshot#676]
I have no sound when I export a video using openshot itself. I guess it's related to this error:

[openshot-FFmpegWriter-test.log](https://github.com/OpenShot/libopenshot/files/6530137/openshot-FFmpegWriter-test.log)
```
/home/stefan/projects/libopenshot/tests/FFmpegWriter.cpp:92
...............................................................................

/home/stefan/projects/libopenshot/tests/FFmpegWriter.cpp:121: FAILED:
  CHECK( r1.info.has_audio == true )
with expansion:
  false == true

/home/stefan/projects/libopenshot/tests/FFmpegWriter.cpp:124: FAILED:
  CHECK( r1.GetFrame(1)->GetAudioChannelsCount() == 2 )
with expansion:
  0 == 2

/home/stefan/projects/libopenshot/tests/FFmpegWriter.cpp:125: FAILED:
  CHECK( r1.info.channel_layout == LAYOUT_STEREO )
with expansion:
  4 == 3

===============================================================================
test cases:  2 |  1 passed | 1 failed
assertions: 15 | 12 passed | 3 failed
```

I have tracked down the issue to this change: https://github.com/OpenShot/libopenshot/commit/1dba02f5b01fb0aefe30a2286f0507839ad0ecc9

If I revert it (or the entire PR #653, it has to resolve some conflicts) the test passes.

**Comment by steils** [comment]
Yes, I'm using the latest snapshots:
libopenshot-audio: 92ff6084d3a773ae2866d786c3107ff6973ceee1
libopenshot: b09416c2e3979246c788abfcf632dd36ab7da046

**Comment by jonoomph** [comment]
Yes, this is an issue with FFmpeg 4, and some upstream changes to FFmpeg that have broken audio encoding in libopenshot. Can you confirm your version of FFmpeg you are using?

**Comment by steils** [comment]
Yes, 4.4.

**Comment by Apteryks** [comment]
Perhaps useful information for a fix: https://devtalk.blender.org/t/newest-ffmpeg-v4-4-is-having-issues-with-blender-2-92-animation-audio-is-not-rendered-with-video/18663/3

**Comment by jonoomph** [comment]
Thanks for troubleshooting this a bit! Appreciate it. https://github.com/OpenShot/libopenshot/commit/1dba02f5b01fb0aefe30a2286f0507839ad0ecc9 Was actually solving an issue with older FFmpeg versions. Okay, this helps, and I need to setup a good VM to test this on different FFmpeg versions, to ensure a final solution works on the FFmpeg 3 and 4 branches.

**Comment by jonoomph** [comment]
And, if I recall, that same commit was only crashing when using MONO channel layout (and a single channel). In other words, it was related to channels and the value of `nb_samples`. So this sounds like the same issue Blender was facing as well. Very interesting. I feel like we are getting closer.

**Comment by lordscales91** [comment]
Any news regarding this issue? I tried to run Example.py from examples in Windows 10 on a MinGW64 MSYS shell and the output video didn't have audio.

**Comment by jonoomph** [comment]
This might take a few days before it makes it downstream into package managers. But I feel pretty confident once this change makes it to your FFmpeg 4 systems, all will work fine! :1st_place_medal:

**Comment by lfam** [comment]
Hi,

I noticed that the fix in commit db7f0fb5e7d36bdd7d15221720b6123fad76820e doesn't apply to libopenshot v0.2.5. Is there a fix that downstream packagers should use for that version?
